cmd/thanos/sidecar.go

```diff
@@ -68,6 +68,9 @@ func registerSidecar(m map[string]setupFunc, app *kingpin.Application, name stri
 	cmd.Flag("s3.insecure", "Whether to use an insecure connection with an S3-Compatible API.").
 		Default("false").Envar("S3_INSECURE").BoolVar(&s3config.Insecure)
 
+	cmd.Flag("s3.signature-version2", "Whether to use S3 Signature Version 2; otherwise Signature Version 4 will be used.").
```

Comment:
ditto

Reply:
Its value is stored in the `*s3config` variable.

The way the flags are defined is inconsistent between Thanos commands for some reason. I started refactoring it, but held off to avoid conflicts with other PRs I have open.

Reply:
yea, can see now, sorry.